packages/web-components/fast-element/src/styles/element-styles.ts

```diff
@@ -147,102 +175,81 @@ function reduceBehaviors(
+                sheet = new CSSStyleSheet();
+                (sheet as any).replaceSync(x);
+                styleSheetCache.set(x, sheet);
+            }
+
+            return sheet;
+        });
     }
 
     public addStylesTo(target: StyleTarget): void {
-        target.adoptedStyleSheets = [...target.adoptedStyleSheets!, ...this.styleSheets];
-        super.addStylesTo(target);
+        target.adoptedStyleSheets = [...target.adoptedStyleSheets!, ...this.sheets];
     }
 
     public removeStylesFrom(target: StyleTarget): void {
-        const sourceSheets = this.styleSheets;
+        const sheets = this.sheets;
         target.adoptedStyleSheets = target.adoptedStyleSheets!.filter(
-            (x: CSSStyleSheet) => sourceSheets.indexOf(x) === -1
+            (x: CSSStyleSheet) => sheets.indexOf(x) === -1
         );
-        super.removeStylesFrom(target);
     }
 }
 
+function usableTarget(target: StyleTarget): StyleTarget {
+    return target === document ? document.body : target;
+}
+
 /**
  * @internal
  */
-export class StyleElementStyles extends ElementStyles {
-    private readonly styleSheets: string[];
+export class StyleElementStrategy implements StyleStrategy {
     private readonly styleClass: string;
 
-    public constructor(styles: ComposableStyles[]) {
-        super(styles, reduceBehaviors(styles));
-        this.styleSheets = reduceStyles(styles) as string[];
+    public constructor(private readonly styles: string[]) {
         this.styleClass = nextId();
     }
 
     public addStylesTo(target: StyleTarget): void {
-        const styleSheets = this.styleSheets;
-        const styleClass = this.styleClass;
+        target = usableTarget(target);
 
-        target = this.normalizeTarget(target);
+        const styles = this.styles;
+        const styleClass = this.styleClass;
 
-        for (let i = 0; i < styleSheets.length; i++) {
+        for (let i = 0; i < styles.length; i++) {
             const element = document.createElement("style");
-            element.innerHTML = styleSheets[i];
+            element.innerHTML = styles[i];
             element.className = styleClass;
             target.append(element);
         }
-
-        super.addStylesTo(target);
     }
 
     public removeStylesFrom(target: StyleTarget): void {
-        target = this.normalizeTarget(target);
-
         const styles: NodeListOf<HTMLStyleElement> = target.querySelectorAll(
             `.${this.styleClass}`
         );
 
+        target = usableTarget(target);
+
         for (let i = 0, ii = styles.length; i < ii; ++i) {
             target.removeChild(styles[i]);
         }
-
-        super.removeStylesFrom(target);
-    }
-
-    public isAttachedTo(target: StyleTarget): boolean {
-        return super.isAttachedTo(this.normalizeTarget(target));
-    }
-
-    private normalizeTarget(target: StyleTarget): StyleTarget {
-        return target === document ? document.body : target;
     }
 }
+
+ElementStyles.setDefaultStrategy(
```

Comment:
Official, clean extensibility API with no hacks for overriding default style handling strategies. We set it up by default to the same as in FASTElement 1.0.